File: test/samples/manualformat/manualformat.hs
import Control.Monad (replicateM)

manualFormat :: Int     -- ^ a first number
             -> Int     -- ^ a second number
             -> Int     -- ^ a third number
             -> Int     -- ^ This one on a new line
             -> [Int]   -- ^ an integer list, one per line
             -> String  -- ^ TODO
-- From the function perspective, this is just 4 integers
manualFormat a b c n onePerLine = "TODO"

main :: IO ()
main = do
  [a, b, c] <- fmap (map read . words) getLine
  n <- fmap read getLine
  onePerLine <- replicateM 3 $ fmap read getLine
  putStrLn $ manualFormat a b c n onePerLine

File: test/samples/manualformat/manualformat.pl
#!/usr/bin/perl
use strict;
use warnings;

# $a_: a first number
# $b_: a second number
# $c: a third number
# $n: This one on a new line
# @one_per_line: an integer list, one per line
sub manual_format {
    my ($a_, $b_, $c, $n, $one_per_line) = @_;
    # TODO From the function perspective, this is just 4 integers
}

my @words = split / /, <>;
my $a_ = int($words[0]);
my $b_ = int($words[1]);
my $c = int($words[2]);
my $n = int <>;
my @one_per_line = ();
for (1..3) {
    push(@one_per_line, int <>);
}

manual_format($a_, $b_, $c, $n, \@one_per_line);
